Teja tmb qc

tkoganti opened the pull request:
## Purpose/implementation Section
1. Adding coverage table that shows differences between old and new TCGA BED files 
2. Adding a .sh scripts that takes in hg19 BED files and converts coordinates to Gh38

#### What scientific question is your analysis addressing?
Coverage data
New. Gh38 BED files 

#### What was your approach?
1. Using `sambamba depth base` calculated number of bases >20x that cover old TCGA BED (gencode) and new agilent TCGA target capture BED 
2. Using CrossMap 

#### What GitHub issue does your pull request address?
https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/597


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
Adding the coverage table for 160 samples in results


### Results

#### What types of results are included (e.g., table, figure)?
`analyses/tcga-capture-kit-investigation/results/TCGA_oldandnew_coverage_comparisons.txt`





#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
-->

Commit:
Adding coverage for TCGA BED

Commit:
Adding target old and new coverage  for TCAG BED

Commit:
Adding script for hg19 to Gh38 using CrossMap

Commit:
git pull all changes YUankun submitted on 03-05-2020
Merge branch 'analyses/tcga-capture-kit-investigation' of https://github.com/yuankunzhu/OpenPBTA-analysis into Teja_tmbQC

yuankunzhu reviewed (commented):
we don't need this file `.ipynb_checkpoints/Untitled-checkpoint.ipynb`, should remove it from this PR
https://github.com/yuankunzhu/OpenPBTA-analysis/pull/2/files?file-filters%5B%5D=.sh#diff-6d2fa397ff3e49c2086cda503a3b3c19

yuankunzhu reviewed (changes requested):
- [x] remove `./ipynb_checkpoints/Untitled-checkpoint.ipynb` from this PR
- [x] git pull to sync up the updates around ci
- [x] update cross-map to make sure we have the chain.gz file in place
- [x] update cross-map output filename so that it will compatible for downstream bedtools intersect run
- [x] update cross-map in `run-investigation.sh`
- [x] test this in docker env

Commit:
Added coverage  plot

Commit:
Added coverage  plot

yuankunzhu commented:
@tkoganti can we put the coverage plot under `plots/` folder?

Commit:
Added crossmap, fixed run_investigation and moved coverage to /plots

Commit:
Removed CrossMapBED.sh script since all of the commands are in run_investigation.sh script

Commit:
Changed WORKING_DIR in run-investigation and all other corresponding paths

Commit:
Changed BED names in intersect-bed-maf.sh

Commit:
Updated  READ.md files

tkoganti commented:
Hi @yuankunzhu I updated the files as requested above. I also updated `scripts/intersect-bed-maf.sh` because that script uses the new BED files. 
Updated README.md as well

Commit:
Updated  READ.md file

yuankunzhu commented on analyses/tcga-capture-kit-investigation/README.md:
```diff
@@ -41,10 +41,9 @@ We created script to prepare the input dataframe for the boxplot script. This sc
-| TCGA-HT-7684-01A-11D-2253-08 | 27755836                     | 84%                          | 8787516                     | 31%                         | 
-| TCGA-SP-A6QF-01A-12D-A35I-08 | 27350549                     | 82%                          | 30060356                    | 31%                         | 
+### Coverage-Comparison
+In order to show in the form of coverage that the downloaded BED files were  correct, we calculated the number of bases that are covered  at atleast 20x in both old TCGA BED file(`gencode.v19.basic.exome`) and the new downloaded BED files (`results/*Gh38.bed`). The coverage table in `results/TCGA_oldandnew_coverage_comparisons.txt` gives  a  tabular summary of  number of bases and percentages wit ...
```
Did you calculated the coverage based on the actual BED for each samples? how did you handle those multiple-capture-kit `|` delimited samples?

also add a space for "atleast"?

tkoganti replied:
These were submitted yesterday before we had the file with two capture kit BED's for the 5 samples. I am doing the BED intersection now, I can re-submit the 5 samples and update. these results.

tkoganti replied:
As per our discussion, since I am using the larger BED file to calculate these coverages here, we are going to leave these for now.

yuankunzhu commented on analyses/tcga-capture-kit-investigation/README.md:
```diff
@@ -41,10 +41,9 @@ We created script to prepare the input dataframe for the boxplot script. This sc
-| TCGA-HT-7684-01A-11D-2253-08 | 27755836                     | 84%                          | 8787516                     | 31%                         | 
-| TCGA-SP-A6QF-01A-12D-A35I-08 | 27350549                     | 82%                          | 30060356                    | 31%                         | 
+### Coverage-Comparison
+In order to show in the form of coverage that the downloaded BED files were  correct, we calculated the number of bases that are covered  at atleast 20x in both old TCGA BED file(`gencode.v19.basic.exome`) and the new downloaded BED files (`results/*Gh38.bed`). The coverage table in `results/TCGA_oldandnew_coverage_comparisons.txt` gives  a  tabular summary of  number of bases and percentages wit ...
```
did we mention how the `results/TCGA_oldandnew_coverage_comparisons.txt` was generated? maybe a quick sambaba command line? also, i'm thinking to change the file extension to tsv, so that it can be better previewed on github?

tkoganti replied:
I changed the file extension. I am going to include the sambamba command. Which  folder would you like the sambbamba command file to go?

yuankunzhu commented on analyses/tcga-capture-kit-investigation/run-investigation.sh:
```diff
@@ -2,62 +2,37 @@
 do 
     filename=`basename $i`
-    curl -s $i | awk '{print "chr"$0}' > scratch/$filename
+    curl -s $i | awk '{print "chr"$0}' > ../../scratch/$filename
```
have you tested this in the container for this? will the `../../scratch/` be compatible with the ci run? https://github.com/tkoganti/OpenPBTA-analysis/blob/Teja_tmbQC/.circleci/config.yml#L209

tkoganti replied:
This works according to the .sh script in ependymoma [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-EPN/run-molecular-subtyping-EPN.sh) Let me know if you want me to change these paths back to what they were in your original script

Commit:
Updated READ.md

Commit:
CHanging coverage file extension from .txt  to .tsv

yuankunzhu commented:
@tkoganti i did some updates on this branch, would you pull and sync up with my changes and solved the conflicts for this PR? thanks

Commit:
Resolved conflicts in README.md

yuankunzhu commented on analyses/tcga-capture-kit-investigation/README.md:
```diff
@@ -21,14 +21,16 @@ Since we got the TCGA data manifest from the [GDC portal](https://portal.gdc.can
+By checking the `results/tcga-capture_kit-info.csv`, it turns out there are BAMs with `|` in the returned capture kit name and url which are those with more than one capture kit which neither the GDC nor its origin data center could retrieve/figure out what the actual capture kit had been applied. We plan to generate an intersected BED for those samples and used that for our analysis. We created  ...
 
 - script: `run-investigation.sh` and `scripts/prepare-tcga-capture_kit.sh`
 - notebook: [`02-prepare-tcga-capture_kit.ipynb`](./notebooks/02-prepare-tcga-capture_kit.ipynb)
```
think i removed all notebooks from my branch, did you sync up on that?

yuankunzhu committed:
Update README.md

yuankunzhu committed:
Update run-investigation.sh

tkoganti committed:
Update README.md
